index.js

```diff
@@ -156,7 +156,9 @@ Parser.prototype._online = function (buf, start, end) {
   }
 
   if (this.strict && cells.length !== this.headers.length) {
-    this.emit('error', new Error('Row length does not match headers'))
+    var e = new Error('Row length does not match headers')
+    e.name = 'SyntaxError'
```

Comment:
I don't believe that `SyntaxError` is correct here. According to the Node docs: https://nodejs.org/api/errors.html#errors_class_syntaxerror, this is used only for invalid JavaScript , or as a result of code evaluation.

`RangeError` should be more appropriate https://nodejs.org/api/errors.html#errors_class_rangeerror. And it's a built-in, so no need to set the `name` property, just instantiate directly.